cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2040,10 +2040,17 @@ macro(build_jemalloc)
     # Enable jemalloc debug checks when Arrow itself has debugging enabled
     list(APPEND JEMALLOC_CONFIGURE_COMMAND "--enable-debug")
   endif()
+
   set(JEMALLOC_BUILD_COMMAND ${MAKE} ${MAKE_BUILD_ARGS})
+  # Paralleism for Make fails with CMake > 3.28 see #39517
+  if(${CMAKE_GENERATOR} MATCHES "Makefiles")
+    list(APPEND JEMALLOC_BUILD_COMMAND "-j1")
```

Comment:
Why do we want to use `-j1` only for jemalloc?
It seems that this problem may be happen with other software. So how about setting `MAKE_BUILD_ARGS`?

Reply:
The problem has not occurred with any other dependency neither on CI nor in my own testing (or after the fix). For the smaller dependencies it might not make a difference but with the number of larger dependencies we have forcing the all to use -j1 if there is no actual issue seems like it would slow build times with no gain?

Reply:
It's not a strong opinion but I think that we should choose safer option (disable `make`'s parallel build for all dependencies) here.
Because we have a workaround (use `-GNinja`) for the parallel build problem.

Reply:
We only have 2 cores available on cran (policy), no ninja (afaik) and our  build times are already being criticized so....

Reply:
If we have only 2 cores, we should not use parallel build for dependencies.
If we use parallel build, it will use 2 more cores in total and slow down total build time...

Reply:
FYI: We use `make` only for jemalloc, bzip2 and UCX.